internal/jennies/golang/strictjson.go

```diff
@@ -0,0 +1,97 @@
+package golang
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/grafana/cog/internal/ast"
+	"github.com/grafana/cog/internal/jennies/common"
+	"github.com/grafana/cog/internal/jennies/template"
+	"github.com/grafana/cog/internal/languages"
+)
+
+type strictJSONUnmarshal struct {
+	tmpl          *template.Template
+	imports       *common.DirectImportMap
+	packageMapper func(string) string
+	typeFormatter *typeFormatter
+}
+
+func newStrictJSONUnmarshal(tmpl *template.Template, imports *common.DirectImportMap, packageMapper func(string) string, typeFormatter *typeFormatter) strictJSONUnmarshal {
+	return strictJSONUnmarshal{
+		tmpl: tmpl.Funcs(template.FuncMap{
+			"formatType": typeFormatter.formatType,
+			"importStdPkg": func(pkg string) string {
+				return imports.Add(pkg, pkg)
+			},
+			"importPkg": packageMapper,
+		}),
+		imports:       imports,
+		packageMapper: packageMapper,
+		typeFormatter: typeFormatter,
+	}
+}
+
+func (jenny strictJSONUnmarshal) generateForObject(buffer *strings.Builder, context languages.Context, object ast.Object) error {
+	if !jenny.objectNeedsUnmarshal(object) {
+		return nil
+	}
+
+	jsonUnmarshal, err := jenny.renderUnmarshal(context, object)
+	if err != nil {
+		return err
+	}
+
+	buffer.WriteString(jsonUnmarshal)
+	buffer.WriteString("\n")
+
+	return nil
+}
+
+func (jenny strictJSONUnmarshal) objectNeedsUnmarshal(obj ast.Object) bool {
+	return obj.Type.IsStruct()
+}
+
+func (jenny strictJSONUnmarshal) renderUnmarshal(context languages.Context, obj ast.Object) (string, error) {
+	tmpl := jenny.tmpl.
+		Funcs(common.TypeResolvingTemplateHelpers(context)).
+		Funcs(template.FuncMap{
+			"resolvesToArrayOfScalars": func(typeDef ast.Type) bool {
```

Comment:
Could be move these ones into the common helpers?

Reply:
Not really: I cheated and included enums in what these functions consider to be scalars (which is valid in Go but not necessarily in other languages)